importers/djornl/parser.py

```diff
@@ -0,0 +1,213 @@
+"""
+Loads the Dan Jacobson/ORNL group's gene and phenotype network data into
+arangodb.
+
+Running this requires a set of source files provided by the ORNL group.
+"""
+import json
+import requests
+import os
+import csv
+
+import importers.utils.config as config
+
+
+class DJORNL_Parser(object):
+
+    def config(self):
+        if not hasattr(self, '_config'):
+            return self._configure()
+
+        return self._config
+
+    def _configure(self):
+
+        configuration = config.load_from_env(extra_required=['ROOT_DATA_PATH'])
+
+        # Collection name config
+        configuration['_NODE_NAME'] = 'djornl_node'
+        configuration['_EDGE_NAME'] = 'djornl_edge'
+
+        # Path config
+        configuration['_NODE_PATH'] = os.path.join(
+            configuration['ROOT_DATA_PATH'],
+            'aranet2-aragwas-MERGED-AMW-v2_091319_nodeTable.csv'
+        )
+        configuration['_NODE_FILE_COL_COUNT'] = 20
+
+        configuration['_EDGE_PATH'] = os.path.join(
+            configuration['ROOT_DATA_PATH'],
+            'merged_edges-AMW-060820_AF.tsv'
+        )
+        configuration['_EDGE_FILE_COL_COUNT'] = 5
+
+        _CLUSTER_BASE = os.path.join(configuration['ROOT_DATA_PATH'], 'cluster_data')
+        configuration['_CLUSTER_PATHS'] = {
+            'cluster_I2': os.path.join(
+                _CLUSTER_BASE,
+                'out.aranetv2_subnet_AT-CX_top10percent_anno_AF_082919.abc.I2_named.tsv'
+            ),
+            'cluster_I4': os.path.join(
+                _CLUSTER_BASE,
+                'out.aranetv2_subnet_AT-CX_top10percent_anno_AF_082919.abc.I4_named.tsv'
+            ),
+            'cluster_I6': os.path.join(
+                _CLUSTER_BASE,
+                'out.aranetv2_subnet_AT-CX_top10percent_anno_AF_082919.abc.I6_named.tsv'
+            ),
+        }
+        self._config = configuration
+        return self._config
+
+
+    def load_edges(self):
+        # Headers and sample row:
+        # node1	node2	edge	edge_descrip	layer_descrip
+        # AT1G01370	AT1G57820	4.40001558779779	AraNetv2_log-likelihood-score	AraNetv2-LC_lit-curated-ppi
+        edge_remap = {
+          'AraGWAS-Phenotype_Associations':         'pheno_assn',
+          'AraNetv2-CX_pairwise-gene-coexpression': 'gene_coexpr',
+          'AraNetv2-DC_domain-co-occurrence':       'domain_co_occur',
+          'AraNetv2-HT_high-throughput-ppi':        'ppi_hithru',
+          'AraNetv2-LC_lit-curated-ppi':            'ppi_liter',
+        }
+
+        node_ix = {}
```

Comment:
ix = abbreviation for index

Reply:
Okay, so it serves as a set of node keys?
My understanding is it will make sure to save a node document referenced by every edge, even if that node does not have any entry in another source file.